### PR TITLE
KTLN-785 - Thread Safe Data Structures in Kotlin

### PR DESCRIPTION
**Pull request by hangga**

### CODE REVIEW

File: core-kotlin-modules/core-kotlin-concurrency-4/src/test/kotlin/com/baeldung/threadsafe/ThreadSafeUnitTest.kt [hunk 1]
```diff
@@ -0,0 +1,298 @@
...
+    }
+
+    @Test
+    fun `test using Collections-synchronizedMap to prevent thread-safety issue`() {
+        val map = Collections.synchronizedMap(HashMap<Int, String>())
+
+        thread {
+            for (i in 1..100) {
+                map[i] = "Thread 1 - $i"
+                Thread.sleep(100) // simulate delay
```
Comment (theangrydev) [inline core-kotlin-modules/core-kotlin-concurrency-4/src/test/kotlin/com/baeldung/threadsafe/ThreadSafeUnitTest.kt:253]: Reduce the delay a bit so the test runs faster

File: core-kotlin-modules/core-kotlin-concurrency-4/src/test/kotlin/com/baeldung/threadsafe/ThreadSafeUnitTest.kt [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+            }
+        }.join()
+
+        thread { map.remove(200) }.join()
+
+        assertEquals(199, map.size)
+    }
+
+    @Test
+    fun `test using ConcurrentHashMap for alternative to Collections-synchronizedMap`() {
```
Comment (theangrydev) [inline core-kotlin-modules/core-kotlin-concurrency-4/src/test/kotlin/com/baeldung/threadsafe/ThreadSafeUnitTest.kt:269]: This one doesn't make sense, if we are joining each time then it's basically single-threaded + it should be possible to modify concurrently with a ConcurrentHashMap without getting a ConcurrentModificationException

File: core-kotlin-modules/core-kotlin-concurrency-4/src/test/kotlin/com/baeldung/threadsafe/ThreadSafeUnitTest.kt [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+        for (item in list) {
+            if (item == 3) {
+                list.remove(item)
+            }
+        }
+    }
+
+    @Test
+    fun `test using AtomicInteger to prevent race-condition`() {
```
[the remark this answers is not in the document]
Comment (hangga) [reply core-kotlin-modules/core-kotlin-concurrency-4/src/test/kotlin/com/baeldung/threadsafe/ThreadSafeUnitTest.kt:186]: Yeah, I think I'll make another scenario for this atomic.

**Comment by hangga** [comment]
Hi @theangrydev I made some changes here. After this, I will update the article. (if you have approved this pr)

**Pull request closed without merging**